osu.Game/Screens/Select/FilterCriteria.cs

```diff
@@ -114,6 +114,27 @@ public string SearchText
 
         public IRulesetFilterCriteria? RulesetCriteria { get; set; }
 
+        public struct OptionalArray<T> : IEquatable<OptionalArray<T>>
+            where T : struct
+        {
+            public bool HasFilter => Values?.Length > 0;
+
+            public bool IsInRange(T value)
+            {
+                return Values?.Contains(value) ?? false;
+            }
+
+            public T[]? Values;
```

Comment:
The null-handling here makes things harder to understand imo. I'd just initialise to empty array by default and thus stop handling nulls.

Better yet, make it an empty `List<T>` so that the usage doesn't have to linq-append-then-reallocate-array every time an item is added here. Yes it realistically probably doesn't matter at all but it just feels sloppy imo.